storage/util/dynamic_percentile.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2024 Google LLC
```

Comment:
Let's 1. move all this code to files in the top level storage package and 2. make all types unexported. I don't think it's necessary to add a separate utils package just for this, and I don't want external users depending on this code.